apollo-router/src/configuration/expansion.rs

```diff
@@ -98,31 +147,46 @@ impl Expansion {
         }
     }
 
-    pub(crate) fn expand_env_variables(
+    pub(crate) fn expand(
         &self,
         configuration: &serde_json::Value,
     ) -> Result<serde_json::Value, ConfigurationError> {
         let mut configuration = configuration.clone();
-        self.env_defaults(&mut configuration)?;
+        self.defaults(&mut configuration)?;
         self.visit(&mut configuration)?;
         Ok(configuration)
     }
 
-    fn env_defaults(&self, config: &mut Value) -> Result<(), ConfigurationError> {
+    fn defaults(&self, config: &mut Value) -> Result<(), ConfigurationError> {
         // Anything that needs expanding via env variable should be placed here. Don't pollute the codebase with calls to std::env.
         // For testing we have the one fixed expansion. We don't actually want to expand env variables during tests
         let mut transformer_builder = TransformBuilder::default();
         transformer_builder =
             transformer_builder.add_action(Parser::parse("", "").expect("migration must be valid"));
         for default in &self.defaults {
-            let env_variable =
-                std::env::var(&default.env_name).unwrap_or_else(|_| default.default.clone());
-            if jsonpath_lib::select(config, &format!("$.{}", default.config_path))
-                .unwrap_or_default()
-                .is_empty()
+            let value = if let Some(env_name) = &default.env_name {
+                std::env::var(env_name).unwrap_or_else(|_| default.default.clone())
+            } else {
+                default.default.clone()
+            };
+
+            if default.env_name.is_some()
+                && !jsonpath_lib::select(config, &format!("$.{}", default.config_path))
+                    .unwrap_or_default()
+                    .is_empty()
             {
+                // Env variables do NOT override config. If the user want to enable this then they must use config expansion.
+                continue;
+            }
+
+            if default.quote {
```

Comment:
I don't understand what `quote` is and what it represents/does :/

Reply:
I had a better idea in the end [8901a4a](https://github.com/apollographql/router/pull/2490/commits/8901a4a7176a2019c09d38c21d36ab74bfb14144)

Buildstructor gives us nice Into methods, so instead of quoting we can try and coerce env variables into the desired type. If it fails then it falls back to string.